### ISSUE TITLE
Exceeding tokenlimit leads to loop

### ISSUE THREAD

**Issue by TheRealVidex** [Significant-Gravitas/Auto-GPT#2590]
### ⚠️ Search for existing issues first ⚠️

- [X] I have searched the existing issues, and there is no existing issue for my problem

### Which Operating System are you using?

Windows; current stable version

### GPT-3 or GPT-4?

GPT-3.5

### Steps to reproduce 🕹

I tryed to setup AutoGPT as an ebook writer. The handling of much text in the process leads to a loop by overstepping the tokenlimit.

### Current behavior 😯

NEXT ACTION:  COMMAND = message_agent ARGUMENTS = {'key': 0, 'message': 'Please generate a list of potential keywords and phrases that I can use to optimize the ebook for search engines.'}
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for ...
Input:y
-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-=
SYSTEM:  Command message_agent returned: Error: This model's maximum context length is 4097 tokens. However, your messages resulted in 4260 tokens. Please reduce the length of the messages.

### Expected behavior 🤔

Handing over the correct amount of tokens (.tokenlimit is set to 4k in .env).

### Your prompt 📝

```yaml
# Paste your prompt here
```

### Your Logs 📒

```log
<NEXT ACTION:  COMMAND = message_agent ARGUMENTS = {'key': 0, 'message': 'Please generate a list of potential keywords and phrases that I can use to optimize the ebook for search engines.'}
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for ...
Input:y
-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-=
SYSTEM:  Command message_agent returned: Error: This model's maximum context length is 4097 tokens. However, your messages resulted in 4260 tokens. Please reduce the length of the messages.>
```

**Comment by Drlordbasil** [comment]
I'm running into same, we need to limit certain chunks I think. Should be able to change chunk size to fix, not sure if that'll fix the total token amount, but we can make the summaries smaller too.

**Comment by Drlordbasil** [comment]
I'll let you know if i run into it after making that change.

**Comment by Drlordbasil** [comment]
My issue is usually generated by the browse function so im changing this:
        self.browse_chunk_max_length = int(os.getenv("BROWSE_CHUNK_MAX_LENGTH", 8192))
        self.browse_summary_max_token = int(os.getenv("BROWSE_SUMMARY_MAX_TOKEN", 300))

To:
        self.browse_chunk_max_length = int(os.getenv("BROWSE_CHUNK_MAX_LENGTH", 2192))
        self.browse_summary_max_token = int(os.getenv("BROWSE_SUMMARY_MAX_TOKEN", 300))
        
   I know thatll fix my issue, but im working on yours

**Comment by Drlordbasil** [comment]
Maybe try changing this line in autogpt/processing/text.py ?
def split_text(text: str, max_length: int = 8192) -> Generator[str, None, None]:

honestly, I'm still checking to see if that'd be it, but doubtful lol

**Comment by Drlordbasil** [comment]
Basically max is 8192 tokens in this context, lowering that will force it to split something less into chunks IE:

def split_text(text: str, max_length: int = 4192) -> Generator[str, None, None]:

basically would split anything above that I believe. It's linked into messages and other functs

**Comment by Drlordbasil** [comment]
It looks like they are making that same type of fix on the new release coming

**Comment by Drlordbasil** [comment]
#2088

**Comment by TheRealVidex** [comment]
Alright, cant wait to try it, but thanks for your time!

**Comment by Pwuts** [comment]
Should be fixed in #2542 just now

**Comment by swapneils** [comment]
I also just ran into this issue after cloning from master a few hours ago; message_agent went over the limit once, after which subsequent calls also failed. 

![image](https://user-images.githubusercontent.com/32959209/235833434-4b3f1bfb-7352-4a72-9ec0-dd5ae7b567c8.png)
